---
layout: post
title: getting ready for Fst and PBS
tags: india
---


Just opened up the India filter file and it is apparently 1016 individuals rather than the 1171 in previous post. I think the biggest differences come from taking out a bunch of southern Europeans, after commiting last post, but honestly cannot remember. 

Also of note: made a switch to plink 1.9, which is way faster and provides the same information in much more R-friendly tables. Have tested some of the old commands like `--keep-allele-order` and they work as expected, although `--freqx` did not seem to respect `--geno` based filtering, so I made a list of SNPs to keep, see `get_fst_in.sh` for more details. Forgot to take out sex chromosomes first time around, so will have to redo everything tomorrow. 

Also checked the sample groups are somewhat sensible:

```
plink --bfile $bedfile --keep $filter --extract india_snps_filter.txt --pca tabs --out 0_test_pca
```

Plotted in R, it all looks good, but should think more about spread of groups being compared, esp dra and ieu, to make sure they're sensible. 

Finally, tested my own WC'84 Fst script against Geneland's implementation and got identical answers. Since my workflow is optimised for my approach, will keep using that one. 